File: src/main/java/com/github/teamclc/sanguosha/event/Event.java
package com.github.teamclc.sanguosha.event;

import cc.huajistudio.aeb.Cancelable;
import com.github.teamclc.sanguosha.event.cause.Cause;

public class Event implements Cancelable {
    private boolean cancelled = false;
    private final String registryName;
    private Cause cause;

    public Event(Cause cause, String registryName) {
        this.cause = cause;
        this.registryName = registryName;
    }

    public String getRegistryName() {
        return registryName;
    }

    public Cause getCause() {
        return cause;
    }

    public void setCause(Cause cause) {
        this.cause = cause;
    }

    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    public void cancel() {
        cancelled = true;
    }
}

File: src/main/kotlin/com/github/teamclc/sanguosha/event/Event.kt
@file:JvmName("GlobalEventBus")
package com.github.teamclc.sanguosha.event

import cc.huajistudio.aeb.Cancelable
import cc.huajistudio.aeb.EventBus

@JvmField
val globalEventBus = EventBus()

abstract class CancelableEvent: Cancelable {
    private var cancelled = false
    override fun isCancelled(): Boolean = cancelled

    fun cancel() {
        cancelled = true
    }
}